### PR TITLE
Fat 1678 c358136

### PR DESCRIPTION
**Pull request by TetianaParanich**
***
E-to-E Test 4 for Deleting Import Logs: Delete filtered logs from View all page (Smoke)
***
https://foliotest.testrail.io//index.php?/cases/view/358136

### CODE REVIEW

**Review by Javokhir12 — APPROVED** [review]

**Review by penfriend — APPROVED** [review]

File: cypress/support/fragments/data_import/logs/logsViewAll.js [hunk 1]
```diff
@@ -258,4 +278,36 @@ export default {
       return body.jobExecutions[0];
     });
   },
+
+  checkIsViewAllOpened:() => {
```
Comment (MaksimTerekhov) [inline cypress/support/fragments/data_import/logs/logsViewAll.js:282]: let's rename to viewAllIsOpened

File: cypress/integration/data-import/filter-and-delete-import-logs-on-view-all.spec.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import DevTeams from '../../support/dictionary/devTeams';
+
+describe('ui-data-import: EDIFACT file import with creating of new invoice record', () => {
+  const startedDate = new Date();
+  const completedDate = startedDate;
+  // format date as YYYY-MM-DD
+  const formattedStart = DateTools.getFormattedDate({ date: startedDate });
+  // api endpoint expects completedDate increased by 1 day
+  completedDate.setDate(completedDate.getDate() + 1);
+  let user = {};
```
[the remark this answers is not in the document]
Comment (TetianaParanich) [reply cypress/integration/data-import/filter-and-delete-import-logs-on-view-all.spec.js:21]: Done

File: cypress/integration/data-import/filter-and-delete-import-logs-on-view-all.spec.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
+import getRandomPostfix from '../../support/utils/stringTools';
+import TestTypes from '../../support/dictionary/testTypes';
+import permissions from '../../support/dictionary/permissions';
+import DataImport from '../../support/fragments/data_import/dataImport';
+import TopMenu from '../../support/fragments/topMenu';
+import LogsViewAll from '../../support/fragments/data_import/logs/logsViewAll';
+import DateTools from '../../support/utils/dateTools';
+import DeleteDataImportLogsModal from '../../support/fragments/data_import/logs/deleteDataImportLogsModal';
+import Users from '../../support/fragments/users/users';
+import InteractorsTools from '../../support/utils/interactorsTools';
```
[the remark this answers is not in the document]
Comment (TetianaParanich) [reply cypress/integration/data-import/filter-and-delete-import-logs-on-view-all.spec.js:10]: I deleted it

File: cypress/integration/data-import/filter-and-delete-import-logs-on-view-all.spec.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    const nameMarcFileForCreate = `C358136autotestFile.${getRandomPostfix()}.mrc`;
+
+    cy.createTempUser([
+      permissions.moduleDataImportEnabled.gui,
+      permissions.deleteImportLogs.gui
+    ])
+      .then(userProperties => {
+        user = userProperties;
+        cy.login(userProperties.username, userProperties.password, { path: TopMenu.dataImportPath, waiter: DataImport.wailtLoading });
+        for (let i = 0; i < 15; i++) {
```
[the remark this answers is not in the document]
Comment (TetianaParanich) [reply cypress/integration/data-import/filter-and-delete-import-logs-on-view-all.spec.js:33]: Done

File: cypress/integration/data-import/filter-and-delete-import-logs-on-view-all.spec.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+          JobProfiles.runImportFile(nameMarcFileForCreate);
+          DataImport.uploadFile('oneMarcBib.mrc', nameMarcFileForCreate);
+          JobProfiles.searchJobProfileForImport('Default - Create Holdings and SRS MARC Holdings');
+          JobProfiles.runImportFile(nameMarcFileForCreate);
+        }
+      });
+  });
+
+  after(() => {
+    Users.deleteViaApi(user.userId);
```
[the remark this answers is not in the document]
Comment (TetianaParanich) [reply cypress/integration/data-import/filter-and-delete-import-logs-on-view-all.spec.js:46]: I created TODO

File: cypress/support/fragments/data_import/logs/logsViewAll.js [hunk 6]
```diff
@@ -93,9 +110,11 @@ export default {
 
   filterJobsByInventorySingleRecordImports(filter) {
     if (filter === 'Yes') {
-      cy.do(CheckBox({ id: 'clickable-filter-singleRecordImports-yes' }).click());
+      cy.do(Accordion('Inventory single record imports')
+        .find(Checkbox({ id: 'clickable-filter-singleRecordImports-yes' })).click());
     } else {
-      cy.do(CheckBox({ id: 'clickable-filter-singleRecordImports-no' }).click());
+      cy.do(Accordion('Inventory single record imports')
```
[the remark this answers is not in the document]
Comment (TetianaParanich) [reply cypress/support/fragments/data_import/logs/logsViewAll.js:116]: Done

File: cypress/support/fragments/data_import/logs/logsViewAll.js [hunk 7]
```diff
@@ -160,9 +179,10 @@ export default {
 
   checkByDate({ from, end }) {
     const queryString = UrlParams.getDateQueryString({ from, end });
-    this.getNumberOfMatchedJobs(queryString).then(count => {
+    return this.getNumberOfMatchedJobs(queryString).then(count => {
       // ensure MultiColumnList is filtered by Date
       this.checkRowsCount(count);
+      cy.wrap(count);
```
[the remark this answers is not in the document]
Comment (TetianaParanich) [reply cypress/support/fragments/data_import/logs/logsViewAll.js:185]: I need to return variable value from function and I can't do it by 'return'

File: cypress/support/fragments/data_import/logs/logsViewAll.js (hunk 1 above)
Comment (TetianaParanich) [reply cypress/support/fragments/data_import/logs/logsViewAll.js:282]: Done

File: cypress/support/fragments/data_import/logs/logsViewAll.js [hunk 8]
```diff
@@ -258,4 +278,36 @@ export default {
...
+    }
+  },
+
+  unmarcCheckbox:(index) => {
+    cy.do(MultiColumnList({ id:'list-data-import' })
+      .find(MultiColumnListCell({ row: 0, columnIndex: index }))
+      .find(Checkbox({ checked: true })).click());
+  },
+
+  checkIsCheckmarkAllLogsRemoved:() => {
```
[the remark this answers is not in the document]
Comment (TetianaParanich) [reply cypress/support/fragments/data_import/logs/logsViewAll.js:303]: Done

File: cypress/support/fragments/data_import/logs/logsViewAll.js [hunk 9]
```diff
@@ -258,4 +278,36 @@ export default {
...
+    cy.do(MultiColumnList({ id:'list-data-import' }).find(Checkbox({ name:'selected-all', checked: false })).exists());
+  },
+
+  deleteLog:() => {
+    cy.do(Pane({ id:'pane-results' }).find(Button('Actions')).click());
+    cy.do(Button('Delete selected logs').click());
+  },
+
+  checkIsModalAbsent:() => { cy.expect(Modal('Delete data import logs?').absent()); }
```
[the remark this answers is not in the document]
Comment (TetianaParanich) [reply cypress/support/fragments/data_import/logs/logsViewAll.js]: Done

**Review by MaksimTerekhov — APPROVED** [review]

**Pull request merged**